File: domain/src/main/java/org/openmedstack/domain/IRouteEvents.java
package org.openmedstack.domain;

import org.openmedstack.events.BaseEvent;

import java.util.concurrent.CompletableFuture;
import java.util.function.Consumer;
import java.util.function.Function;

public interface IRouteEvents {
    <T extends BaseEvent> void register(Consumer<T> handler);

    CompletableFuture dispatch(Object eventMessage) throws HandlerForDomainEventNotFoundException;
}

File: domain/src/main/kotlin/org/openmedstack/domain/IRouteEvents.kt
package org.openmedstack.domain

import org.openmedstack.events.BaseEvent
import java.util.concurrent.CompletableFuture
import java.util.function.Consumer

interface IRouteEvents {
    fun <T : BaseEvent> register(handler: Consumer<T>)
    @Throws(HandlerForDomainEventNotFoundException::class)
    fun dispatch(eventMessage: Any): CompletableFuture<*>
}
